File: code-coaches/easy/ballpark-orders/ballpark_orders.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <numeric>
#include <iomanip>
#include <cmath>

double get_total_cost(std::vector<std::string> order)
{
    std::unordered_map<std::string, double> menu = {
        {"Pizza", 6.0},
        {"Nachos", 6.0},
        {"Cheeseburger", 10.0},
        {"Water", 4.0},
        {"Coke", 5.0},
    };
    double subtotal = std::accumulate(order.begin(), order.end(), 0.0, [&](double sum, const std::string& item) {
        return sum + menu[item] ? menu[item] : 5.0;
    });
    double tax = 1.07;
    double total = subtotal * tax;

    return round(total * 100) / 100;
}

int main()
{
    std::string order_input;
    std::getline(std::cin, order_input);

    std::vector<std::string> order;
    std::istringstream iss(order_input);
    for (std::string item; iss >> item; )
    {
        order.push_back(item);
    }

    std::cout << get_total_cost(order);

    return 0;
}

File: code-coaches/easy/ballpark-orders/BallparkOrders.java
import java.util.*;

public class BallparkOrders {

    public static double getTotalCost(List<String> order) {
        Map<String, Double> menu = new HashMap<String, Double>() {
            {
                put("Pizza", 6.0);
                put("Nachos", 6.0);
                put("Cheeseburger", 10.0);
                put("Water", 4.0);
                put("Coke", 5.0);
            }
        };

        double subtotal = order.stream()
            .mapToDouble(item -> menu.getOrDefault(item, 5.0))
            .sum();
        double tax = 1.07;
        double total = subtotal * tax;

        return Math.round(total * 100.0) / 100.0; // round to 2 decimals
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> order = Arrays.asList(scanner.nextLine().split(" "));

        System.out.println(getTotalCost(order));

        scanner.close();
    }

}
